src/Events/Custom_Tables/V1/Migration/State.php

```diff
@@ -349,4 +349,17 @@ public function save() {
 		] );
 		update_option( self::STATE_OPTION_KEY, $this->data );
 	}
+
+	/**
+	 * Returns whether the current phase is a migration dry-run or not.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool Whether the current phase is a migration dry-run or not.
+	 */
+	public function is_dry_run(): bool {
+		$phase = $this->get_phase();
+
+		return $phase === self::PHASE_PREVIEW_IN_PROGRESS;
```

Comment:
We may want to reference a few more states here, off the top of my head I think these are technically where we go back to a dry run.

`PHASE_REVERT_COMPLETE, PHASE_PREVIEW_PROMPT, PHASE_CANCEL_COMPLETE` - not 100% positive, going off my memory.

Reply:
Since this check is in the State object, the context of `is_migrating` shouldn't be assumed